app/models/item.rb

```diff
@@ -1,2 +1,14 @@
 class Item < ActiveRecord::Base
+  belongs_to :categories
+
+  def self.create_with_category(attrs)
+    create(
+      name: attrs.fetch(:name),
+      category_id: Category.find_or_create_by(name: attrs.fetch(:category, "")).id
```

Comment:
I think I'm good with the `N+1` here. Although I'm wondering if we should create categories that doesn't exist or to define the categories elsewhere and just add them to the product, and don't save if the category doesn't exist. I'm not 100% sure about what's the specification on that particular use case.